### PR TITLE
sidebar menu

### PR DESCRIPTION
**Pull request by qualv13**
sidebar menu with dropdown element

### CODE REVIEW

**Comment by qualv13** [comment]
what do you think

File: index.html [hunk 1]
```diff
@@ -3,11 +3,66 @@
   <head>
     <meta charset="UTF-8" />
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <link rel="stylesheet" href="style.css">
```
Comment (Frendzlu) [inline index.html:6]: By definition, index.html should not be changed. The way Svelte works, it renders the contents of the app tree (starting from a specified file, currently `App.svelte`) by injecting it via `main.ts`

All changes are to be moved to `./src/lib/Sidebar`, and a `<Sidebar>` component should be added in `App.svelte`

File: style.css [hunk 2]
```diff
@@ -0,0 +1,80 @@
+body {
```
Comment (Frendzlu) [inline style.css:1]: `style.css` should be moved into the `Sidebar.svelte` component

File: index.html [hunk 3]
```diff
@@ -3,11 +3,66 @@
...
+
+  
   <body>
+
+    <div class="sidenav">
+      <a href="#profile">Profile</a>
+      <a href="#game">Game</a>
+      <a href="#Leadboard">Leadboard</a>
+      <a href="#contact">Contact</a>
+      <button class="dropdown-btn">Dropdown
```
Comment (Frendzlu) [inline index.html:23]: Change anything not requiring a class (singular instance of name) to id
Remember to change `getElementsByClassName()` to `getElementById()`

File: index.html [hunk 4]
```diff
@@ -3,11 +3,66 @@
...
     <title>Vite + Svelte + TS</title>
   </head>
+
+  
+
+
+  
   <body>
+
+    <div class="sidenav">
```
Comment (Frendzlu) [inline index.html:18]: Sidenav should be collapsible. To do so, you could add a `<button id="toggleSidebar">` with appropriate css (`display: fixed`, use `top`, `left` to position), and make it change the visibility of the entire `sidenav`

File: index.html [hunk 5]
```diff
@@ -3,11 +3,66 @@
...
+    <div class="main">
+      <h2>Does_someone_know_python</h2>
+      <p>Click on the dropdown button to open the dropdown menu inside the side navigation.</p>
+      <p>This sidebar is of full height (100%) and always shown.</p>
+      <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Possimus ab animi, autem enim quas nobis! Commodi, modi ducimus? Quisquam dignissimos fuga harum tempore earum blanditiis amet assumenda ex temporibus rem.</p>
+    </div>
+
+    <script>
+      /* Loop through all dropdown buttons to toggle between hiding and showing its dropdown content - This allows the user to have multiple dropdowns without any conflict */
+      var dropdown = document.getElementsByClassName("dropdown-btn");
```
Comment (Frendzlu) [inline index.html:47]: use `let` instead of `var`, just in case

File: index.html [hunk 6]
```diff
@@ -3,11 +3,66 @@
...
+      var i;
```
Comment (Frendzlu) [inline index.html:48]: Remove: see below

File: index.html [hunk 7]
```diff
@@ -3,11 +3,66 @@
...
+      
+      for (i = 0; i < dropdown.length; i++) {
```
Comment (Frendzlu) [inline index.html:50]: for loops in js/ts should be written like this
```ts
for (let i = 0; i < dropdown.length; i++) {
     //...
}
```
```

File: index.html [hunk 8]
```diff
@@ -3,11 +3,66 @@
...
+        dropdown[i].addEventListener("click", function() {
+        this.classList.toggle("active");
+        var dropdownContent = this.nextElementSibling;
+        if (dropdownContent.style.display === "block") {
+        dropdownContent.style.display = "none";
```
Comment (Frendzlu) [inline index.html:55]: bad formatting, please fix indentation

**Review by Frendzlu — CHANGES_REQUESTED** [review]
Additional info: Component structure should look like so
```svelte
<script lang="ts">
    //imports
    //declarations
    //functions
    //standalone code
</script>
<main>
    <!-- component html structure goes here -->
</main>
<style>
     .sample {
         margin: 0 auto;   
     }
</style> 
```

File: index.html (hunk 3 above)
Comment (qualv13) [reply index.html]: done

**Review by Frendzlu — APPROVED** [review]
lgtm

**Pull request merged**